src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -911,7 +911,7 @@ internal static SourceHashAlgorithm TryParseHashAlgorithmName(string arg)
             string directory = PathUtilities.GetDirectoryName(path);
             string pattern = PathUtilities.GetFileName(path);
 
-            var resolvedDirectoryPath = (directory.Length == 0) ? baseDirectory : FileUtilities.ResolveRelativePath(directory, baseDirectory);
+            var resolvedDirectoryPath = directory?.Length != 0 ? FileUtilities.ResolveRelativePath(directory, baseDirectory) : baseDirectory;
```

Comment:
>directory?.Length != 0 [](start = 40, length = 22)

We'll call `ResolveRelativePath(directory, baseDirectory)` if `directory == null` which seems surprising. Should this be: ```
string.IsNullOrEmpty(directory) ?
    baseDirectory :
    FileUtilities.ResolveRelativePath(directory, baseDirectory)
```